Firestore/core/src/firebase/firestore/auth/firebase_credentials_provider_apple.mm

```diff
@@ -96,7 +96,7 @@
       // Cancel the request since the user changed while the request was
       // outstanding so the response is likely for a previous user (which
       // user, we can't be sure).
-      completion({"", User::Unauthenticated()},
+      completion(Token::Invalid(),
                  "getToken aborted due to user change.");
     } else {
       completion(
```

Comment:
This is the bug in the line below:

token can be `nil` but we're still invoking the constructor that creates a token that appears valid, which then triggered all the rest of the code to behave incorrectly. Your other changes are also reasonable, but I think we have to be considerably more defensive about the values we accept from auth.

I think we should probably move to something like what I did for [`User`](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/auth/user.h#L64). We should construct the right kind of token (vaild or not) given our inputs rather than making the constructor blindly set is_valid to true or false.